api/tests/PoolCandidateTest.php

```diff
@@ -0,0 +1,606 @@
+    });
+
+    // Assert query will with no operationalRequirements filter will return all candidates
+    $this->graphQL(/** @lang Graphql */ '
+      query countPoolCandidates($where: PoolCandidateFilterInput) {
+        countPoolCandidates(where: $where)
+      }
+    ', [
+      'where' => []
+    ])->seeJson([
+      'data' => [
+        'countPoolCandidates' => 7
+      ]
+    ]);
+
+    // Assert query with operationalRequirement filter will return correct candidate count
+    $this->graphQL(/** @lang Graphql */ '
+      query countPoolCandidates($where: PoolCandidateFilterInput) {
+        countPoolCandidates(where: $where)
+      }
+    ', [
+      'where' => [
+        'operationalRequirements' => [[ 'key' => 'new_operational_requirement' ]],
+      ]
+    ])->seeJson([
+      'data' => [
+        'countPoolCandidates' => 2
+      ]
+    ]);
+
+    // Assert query with unknown operationalRequirement filter will return 0
+    $this->graphQL(/** @lang Graphql */ '
+      query countPoolCandidates($where: PoolCandidateFilterInput) {
+        countPoolCandidates(where: $where)
+      }
+    ', [
+      'where' => [
+        'operationalRequirements' => [[ 'key' => 'unknown_operational_requirement' ]],
+      ]
+    ])->seeJson([
+      'data' => [
+        'countPoolCandidates' => 0
+      ]
+    ]);
+  }
+
+  public function testFilterByPool(): void
+  {
+
+    // Create 5 pool candidates.
+    Pool::factory()->count(3)->create();
+    PoolCandidate::factory()->count(5)->create();
+
+    // Create new cmoAsset and attach to 2 new pool candidates.
+    $id = strtolower(preg_replace('/\s+/', '_', 'test'));
+    $pool = Pool::factory()->create();
+    PoolCandidate::factory()->count(2)->create()->each(function($candidate) use ($pool) {
+      $candidate->pool()->associate($pool);
+      $candidate->save();
+    });
+
+    // Assert query will with no cmoAssets filter will return all candidates
+    $this->graphQL(/** @lang Graphql */ '
+      query countPoolCandidates($where: PoolCandidateFilterInput) {
+        countPoolCandidates(where: $where)
+      }
+    ', [
+      'where' => []
+    ])->seeJson([
+      'data' => [
+        'countPoolCandidates' => 7
+      ]
+    ]);
+
+    // Assert query with cmoAsset filter will return correct candidate count
+    $this->graphQL(/** @lang Graphql */ '
+      query countPoolCandidates($where: PoolCandidateFilterInput) {
+        countPoolCandidates(where: $where)
+      }
+    ', [
+      'where' => [
+        'pools' => [[ 'id' => $id ]],
+      ]
+    ])->seeJson([
+      'data' => [
+        'countPoolCandidates' => 2
+      ]
+    ]);
+
+    // Assert query with unknown cmoAsset filter will return 0
+    $this->graphQL(/** @lang Graphql */ '
+      query countPoolCandidates($where: PoolCandidateFilterInput) {
+        countPoolCandidates(where: $where)
+      }
+    ', [
+      'where' => [
+        'pools' => [[ 'id' => 'unknown_id' ]],
```

Comment:
```suggestion
        'pools' => [[ 'id' => '00000000-0000-0000-0000-000000000000' ]],
```
Needs to be a valid UUID to run